discovery-infra/test_infra/helper_classes/kube_helpers.py

```diff
@@ -0,0 +1,376 @@
+            self,
+            api_vip: str = DEFAULT_API_VIP,
+            api_vip_dns_name: str = DEFAULT_API_VIP_DNS_NAME,
+            ingress_vip: str = DEFAULT_INGRESS_VIP,
+            vip_dhcp_allocation: bool = env_variables['vip_dhcp_allocation']
+    ):
+        dict.__init__(
+            self,
+            agentBareMetal={
+                'apiVIP': api_vip,
+                'ingressVIP': ingress_vip,
+                'VIPDHCPAllocation': 'Enabled' if vip_dhcp_allocation else ''
+            }
+        )
+
+        if _does_string_contain_value(api_vip_dns_name):
+            self['agentBareMetal']['apiVIPDNSName'] = api_vip_dns_name,
+
+
+DEFAULT_MACHINE_CIDR = env_variables.get('machine_cidr', '')
+DEFAULT_CLUSTER_CIDR = env_variables.get('cluster_cidr', '172.30.0.0/16')
+DEFAULT_SERVICE_CIDR = env_variables.get('service_cidr', '10.128.0.0/14')
+
+
+class InstallStrategy(dict):
+
+    def __init__(
+            self,
+            control_plane_agents: int = 1,
+            worker_agents: int = 0,
+            label_selector: Optional[Dict[str, str]] = None,
+            host_prefix: int = env_variables['host_prefix'],
+            machine_cidr: str = DEFAULT_MACHINE_CIDR,
+            cluster_cidr: str = DEFAULT_CLUSTER_CIDR,
+            service_cidr: str = DEFAULT_SERVICE_CIDR,
+            ssh_public_key: str = env_variables['ssh_public_key'],
+    ):
+        dict.__init__(
+            self,
+            agent={
+                'networking': {
+                    'clusterNetwork': [{
+                        'cidr': cluster_cidr,
+                        'hostPrefix': host_prefix
+                    }],
+                    'serviceNetwork': [service_cidr]
+                },
+                'provisionRequirements': {
+                    'controlPlaneAgents': control_plane_agents,
+                    'workerAgents': worker_agents,
+                },
+                'agentSelector': {'matchLabels': label_selector or {}}
+            }
+        )
+
+        if _does_string_contain_value(ssh_public_key):
+            self['agent']['sshPublicKey'] = ssh_public_key
+
+        if _does_string_contain_value(machine_cidr):
+            self['agent']['machineNetwork'] = [{'cidr': machine_cidr}],
+
+
+DEFAULT_WAIT_FOR_CRD_STATUS_TIMEOUT = 60
+DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT = 300
+
+
+class ClusterDeployment(_BaseKubeAPIResource):
+    hive_api_group = 'hive.openshift.io'
+
+    def __init__(self, name: str, namespace: str = env_variables['namespace']):
+        _BaseKubeAPIResource.__init__(self, name, namespace)
+
+        self.crd_api = CustomObjectsApi(self.api_client)
+
+        self.__secret = None
+
+    @property
+    def secret(self):
+        return self.__secret
+
+    def create(
+            self,
+            platform: Platform,
+            install_strategy: InstallStrategy,
+            secret: Secret,
+            base_domain: str = env_variables['base_domain'],
+            **kwargs
+    ) -> None:
+
```

Comment:
Please avoid a new blank line at the beginning of functions.

Reply:
why?

Reply:
Although there are no details regarding this behavior on [PEP8](https://www.python.org/dev/peps/pep-0008/#blank-lines) conventions, it just makes the module unnecessary longer.

Reply:
I removed the empty lines although I think it makes it look less convenient to read